plugins/samson_ledger/test/lib/samson_ledger/samson_plugin_test.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+require_relative "../../test_helper"
+
+SingleCov.covered!
+
+describe "the samson_ledger plugin callback" do
+  let(:deploy) { Deploy.first }
+
+  before do
+    stub_request(:post, "https://foo.bar/api/v1/events")
+  end
+
+  with_env(LEDGER_TOKEN: "sometoken", LEDGER_BASE_URL: "https://foo.bar")
+
+  let(:deploy) { Deploy.first }
+
+  it "successfully fires the before_deploy" do
+    assert Samson::Hooks.fire(:before_deploy, deploy, nil)
+  end
+
+  it "successfully fires the after_deploy" do
+    assert Samson::Hooks.fire(:after_deploy, deploy, nil)
+  end
+
+  it "fails to fires the after_deploy" do
```

Comment:
other tests do not assert anything, so I could break the hook and this would still pass ... also will fail when any other hook code changes ...

prefer `Foobar.expects(:post_deploy)` & ``Foobar.expects(:post_deploy).never``